src/Package/Impl/ProjectSystem/Configuration/ProjectConfigurationSettingsProvider.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Diagnostics;
+using Microsoft.R.Components.Application.Configuration;
+using Microsoft.R.Host.Client.Extensions;
+using Microsoft.VisualStudio.ProjectSystem;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.Configuration {
+    /// <summary>
+    /// Provides access to the current project configuration settings collection.
+    /// The collection is shared and there may be multiple concurrent users 
+    /// such as when project property page is open and user also invokes 
+    /// 'Add Database Connection' command. Dispose object to release access
+    /// to the collection. All access is read/write. Do not cache the object.
+    /// </summary>
+    [Export(typeof(IProjectConfigurationSettingsProvider))]
+    internal sealed class ProjectConfigurationSettingsProvider : IProjectConfigurationSettingsProvider {
+        private readonly Dictionary<string, SettingsAccess> _settings = new Dictionary<string, SettingsAccess>();
+        private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
+
+        public Task<IProjectConfigurationSettingsAccess> OpenProjectSettingsAccessAsync(ConfiguredProject configuredProject) {
+            var props = configuredProject.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+            return OpenProjectSettingsAccessAsync(configuredProject.UnconfiguredProject, props);
+        }
+
+        public async Task<IProjectConfigurationSettingsAccess> OpenProjectSettingsAccessAsync(
+            UnconfiguredProject unconfiguredProject, IRProjectProperties propertes) {
+
+            Check.ArgumentNull(nameof(unconfiguredProject), unconfiguredProject);
```

Comment:
Yes, we have it in Core.Diagnostics. It is actually an old class from Blend, haven't seen much use in RTVS though